### ISSUE TITLE
Snapshot Event Handler State

### ISSUE THREAD

**Issue by marpo60** [commanded/commanded#461]
### Context

Event Handler State is provided in the init/1 callback and can be updated in each handle/2
https://hexdocs.pm/commanded/1.3.0/Commanded.Event.Handler.html#module-event-handler-state

This is pretty handy, but as this moment the state is ephemeral, which means that on server restart, there is a need to reconstruct the state.

It would be awesome if the EventHandler state could be persisted in the same fashion as the Process Manager state.

https://github.com/commanded/commanded/blob/1c8af07e4fdc415446b80482bf92f4f9c9873269/lib/commanded/process_managers/process_manager_instance.ex#L82-L84

https://github.com/commanded/commanded/blob/1c8af07e4fdc415446b80482bf92f4f9c9873269/lib/commanded/process_managers/process_manager_instance.ex#L243

If there is value in doing this I can start implementing this.

**Comment by slashdotdash** [comment]
Yes, I agree this would be a useful opt-in feature for event handlers. A pull request would be very welcome, thanks!
